### PR TITLE
Implement Interception Filtering

### PR DESCRIPTION
**Pull request by gdbelvin**
The Trillian Interceptor fails all requests that it doesn't recognize. Unfortunately this doesn't play well when other services are hosted on the same server. 

This PR restricts the Trillian interceptor to operating on Trillian services.

### CODE REVIEW

File: server/interceptor/interceptor.go [hunk 1]
```diff
@@ -258,6 +274,15 @@ func isLeafOK(leaf *trillian.QueuedLogLeaf) bool {
 	return leaf == nil || leaf.Status == nil || leaf.Status.Code == int32(codes.OK)
 }
 
+// serviceName returns "some.package.service" for
+// "/some.package.service/method".
+func serviceName(fullMethod string) string {
+	if !strings.HasPrefix(fullMethod, "/") {
+		return ""
+	}
+	return strings.Split(fullMethod, "/")[1]
```
Comment (AlCutter) [inline server/interceptor/interceptor.go]: This should be ...[0], shouldn't it?

File: server/interceptor/interceptor_test.go [hunk 2]
```diff
@@ -538,24 +577,27 @@ func TestTrillianInterceptor_QuotaInterception_ReturnsTokens(t *testing.T) {
...
 func TestTrillianInterceptor_NotIntercepted(t *testing.T) {
 	tests := []struct {
-		req interface{}
+		method string
+		req    interface{}
 	}{
 		// Admin
-		{req: &trillian.CreateTreeRequest{}},
-		{req: &trillian.ListTreesRequest{}},
+		{method: "/trillian.TrillianAdmin/CreateTree", req: &trillian.CreateTreeRequest{}},
```
Comment (AlCutter) [inline server/interceptor/interceptor_test.go]: Consider extending the test coverage to catch bugs where the interceptor is accidentally universally disabled?

File: server/interceptor/interceptor.go (hunk 1 above)
Comment (gdbelvin) [reply server/interceptor/interceptor.go]: [0] gets us "" because strings start with "/"

File: server/interceptor/interceptor_test.go (hunk 2 above)
Comment (gdbelvin) [reply server/interceptor/interceptor_test.go]: Not sure I follow...
The existing tests verify that the handlers are running for the expected methods. 
If the interceptors get disabled these tests fail when they test for the presence of tree in `ctx`

**Comment by gdbelvin** [comment]
PTAL

File: server/interceptor/interceptor_test.go (hunk 2 above)
Comment (AlCutter) [reply server/interceptor/interceptor_test.go]: ok, then we're good :)

File: server/interceptor/interceptor.go (hunk 1 above)
Comment (AlCutter) [reply server/interceptor/interceptor.go]: Ah, bad assumption on my part. Carry on!

**Pull request merged**